docs/design/storage-auto-scale.md

```diff
@@ -0,0 +1,83 @@
+# Storage Automatic Scaler
+
+## Summary
+
+This feature would help users to get rid of maintenance task of keeping the system with enough available free space.
+In virtualized environments, we can scale the storage devices Osds so this feature helps in automatic scaling. 
+
+## Proposal
+
+Implement auto storage scale CR per device class of a specific Storagecluster.
+
+## High-Level Design
+
+- Create a StorageAutoScale CR per Storagecluster device class
+    ```
+    apiVersion: ocs.openshift.io/v1
+    kind: StorageAutoScaling
+    metadata:
+        name: <storageclusterName>-<deviceClassName>
+        namespace: openshift-storage
+    spec:
+        storageCluster: 
+            name: <storageclusterName>
+        deviceClassName: ssd
+        clusterCapacityLimit: <capacity>
+        maxOsdSize: 8Ti
+        timeoutSeconds: 1800   
+    status:
+        phase: (NotStarted | InProgress | Failed | Succeeded)
+        LastRunTimestamp: (only on Failed | Succeeded)
+        currentOsdSize: 2TiB
```

Comment:
After the expansion is completed (successful), will these "currentOsdSize" values remain in the status? If so, it would be confusing. But if we rename them to "start" osd size and count, then the status still makes sense when it is seen in the status after completion.
```suggestion
        startOsdSize: 2TiB
```